### DISCUSSION TITLE
issue with resize handler

### DISCUSSION THREAD

**Discussion by jayala-gh** [samuelarbibe/dnd-timeline#26] [Q&A] [answered]
I had a problem when I try to resize an item, using the code in the examples I get this result:
![ezgif-6-c16a54bb07](https://github.com/samuelarbibe/dnd-timeline/assets/137443420/e9d8be2b-0587-42cd-9297-faf2b3a8eff1)

when I log relevance when resizing it shows the new values correctly but then it logs the old value, so it sets the prev value back.

dragging items works great.

**Comment by samuelarbibe** [comment, marked as answer]
Hey ayala!
Please provide the source code so that I can help with it.

Anyway, I suggest looking at the documentation here: 
https://samuel-arbibe.gitbook.io/dnd-timeline/documentation/timelinecontext#event-handlers

Please notice this part:  

> Do not try to calculate the item's updated relevance on your own. You should call the `getRelevanceFromDragEvent` or `getRelevanceFromResizeEvent` to get the updated relevance right before using it to update you own state.

Your resize handler code should look something like:  
```tsx
const onResizeEnd = useCallback((event: ResizeEndEvent) => {
  const updatedRelevance = event.active.data.current.getRelevanceFromResizeEvent(event);

  if (!updatedRelevance) return;

  const activeItemId = event.active.id;

  console.log(`Updated ${activeItemId}'s relevance: ${JSON.stringify(updatedRelevance)}`);

  // update your state
}, [...]
```

**Reply by heeba-khan** [reply]
Hey @samuelarbibe ,
You have used getRelevanceFromResizeEvent above in your example but in documentation , you have used getSpanFromResizeEvent. Can you explain a bit how different these are in order to update a span ??

Also the issue I am facing is regarding resizing nested items. Can your library support that ?

**Reply by samuelarbibe** [reply]
This difference is due to a breaking change in the library, described here: https://github.com/samuelarbibe/dnd-timeline/discussions/34

**Reply by samuelarbibe** [reply]
Can you please elaborate on your issue?  
Please open a different discussion if unrelated to the current discussion :)

**Reply by heeba-khan** [reply]
Ok , I got it after reviewing the breaking changes. It make sense now.
Also, I will open another discussion for the issue I am facing. Thanks ;)
